### ISSUE TITLE
Existing IAM role reuse not working

### ISSUE THREAD

**Issue by nerdguru** [serverless/serverless#2707]
# This is a Bug Report

## Description

For bug reports:
* What went wrong?
When placing a role: <arn> at either the Provider or Function level in serverless.yml, the system creates a new IAM role instead of reusing the one specified.

* What did you expect should have happened?
According to the documentation, the system should reuse the existing IAM role

* What was the config you used?
In my serverless.yml:

```
provider:
  name: aws
  runtime: nodejs4.3
  role: arn:aws:iam::<my account number>:role/<some role name>
```
and instead of reusing that IAM role specified, the system creates a new one, here's console output upon deploy:

```
Service Information
service: myFirstServerlessService
 stage: dev
region: us-east-1
api keys:
  None
endpoints:
  None
functions:
  myFirstServerlessService-dev-hello: arn:aws:lambda:us-east-1:<my account number>:function:myFirstServerlessService-dev-hello
```

* What stacktrace or error message from your provider did you see?
No stacktrace or error message, just incorrect behavior.

Similar or dependent issues:

## Additional Data

For more information, here's the forum thread I started on this topic: [Existing IAM role reuse syntax/behavior ](http://forum.serverless.com/t/existing-iam-role-reuse-syntax-behavior/768)
* ***Serverless Framework Version 1.1***:

**Comment by rowanu** [comment]
As mentioned in the forum, this looks like a duplicate of #2706. Other issue points out that it works in `master` (just not v1.1), and I have verified this too (but haven't find the offending/fixing commit).

**Comment by mthenw** [comment]
Fixed in 1.2.

@nerdguru let us know if it's still and issue.
